spec/01-unit/23-vaults_spec.lua

```diff
@@ -0,0 +1,133 @@
+local helpers = require "spec.helpers" -- initializes 'kong' global for vaults
+local conf_loader = require "kong.conf_loader"
+
+
+local is_ref_test_map = {
+  ["{vault://x/x}"] = true,
+  ["{vault://x/xx}"] = true,
+  ["{vault://xx/xx-x}"] = true,
+  ["{vault://xxx/xx-x}"] = true,
+  ["{vault://;/xx-x}"] = true,
+  ["vault:/xx-x}"] = false,
+  ["{vault:/xx-x"] = false,
+  ["vault:/xx-x"] = false,
+  ["{valut:/xx-x}"] = false,
+  ["{vault:/xx-x}"] = false,
+}
+
+
+local get_test_map = {
+  ["{vault://env/test_secrets}"] = {
+    k = "TEST_SECRETS",
+    v = "test_value"
+  },
+  ["{vault://env/test}"] = {
+    k = "TEST",
+    v = ""
+  },
+}
+
+
+describe("Vault PDK", function()
```

Comment:
We usually test PDK methods on the t/ folder. Is there a reason why this was not done there? (we can move those there after feature freeze).

Reply:
We can add some stuff inside PDK tests too.